src/URLCtrl.cpp

```diff
@@ -0,0 +1,236 @@
+//
+//This program is distributed in the hope that it will be useful,
+//but WITHOUT ANY WARRANTY; without even the implied warranty of
+//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//GNU General Public License for more details.
+//
+//You should have received a copy of the GNU General Public License
+//along with this program; if not, write to the Free Software
+//Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+#include "TCHAR.h"
+#include "URLCtrl.h"
+
+/* XOR mask for hand cursor */
+/* Generated by HexEdit */
+static BYTE XORMask[128] =
+{
+  0xff, 0xff, 0xff, 0xff,
+  0xf9, 0xff, 0xff, 0xff,
+  0xf0, 0xff, 0xff, 0xff,
+  0xf0, 0xff, 0xff, 0xff,
+  0xf0, 0xff, 0xff, 0xff,
+  0xf0, 0xff, 0xff, 0xff,
+  0xf0, 0x24, 0xff, 0xff,
+  0xf0, 0x00, 0x7f, 0xff,
+  0xc0, 0x00, 0x7f, 0xff,
+  0x80, 0x00, 0x7f, 0xff,
+  0x80, 0x00, 0x7f, 0xff,
+  0x80, 0x00, 0x7f, 0xff,
+  0x80, 0x00, 0x7f, 0xff,
+  0x80, 0x00, 0x7f, 0xff,
+  0xc0, 0x00, 0x7f, 0xff,
+  0xe0, 0x00, 0x7f, 0xff,
+  0xf0, 0x00, 0xff, 0xff,
+  0xf0, 0x00, 0xff, 0xff,
+  0xf0, 0x00, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+  0xff, 0xff, 0xff, 0xff,
+};
+
+/* AND mask for hand cursor */
+/* Generated by HexEdit */
+static BYTE ANDMask[128] =
+{
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x06, 0x00, 0x00, 0x00,
+  0x06, 0x00, 0x00, 0x00,
+  0x06, 0x00, 0x00, 0x00,
+  0x06, 0x00, 0x00, 0x00,
+  0x06, 0x00, 0x00, 0x00,
+  0x06, 0xdb, 0x00, 0x00,
+  0x06, 0xdb, 0x00, 0x00,
+  0x36, 0xdb, 0x00, 0x00,
+  0x36, 0xdb, 0x00, 0x00,
+  0x37, 0xff, 0x00, 0x00,
+  0x3f, 0xff, 0x00, 0x00,
+  0x3f, 0xff, 0x00, 0x00,
+  0x1f, 0xff, 0x00, 0x00,
+  0x0f, 0xff, 0x00, 0x00,
+  0x07, 0xfe, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+};
+
+
+void URLCtrl::create(HWND itemHandle, LPTSTR link, COLORREF linkColor)
+{
+	// turn on notify style
+	::SetWindowLongPtr(itemHandle, GWL_STYLE, ::GetWindowLongPtr(itemHandle, GWL_STYLE) | SS_NOTIFY);
+
+	// set the URL text (not the display text)
+	if (link)
+		_tcscpy(_URL, link);
+
+	// set the hyperlink colour
+    _linkColor = linkColor;
+
+	// set the visited colour
+	_visitedColor = RGB(128,0,128);
+
+	// subclass the static control
+	_oldproc = (WNDPROC)::SetWindowLongPtr(itemHandle, GWLP_WNDPROC, (LONG)URLCtrlProc);
```

Comment:
Think you could take directly the version from the current n++ trunk or did you do any modification to this file. 

Here and below the cast to LONG is not correct needs to be LONG_PTR, see n++ version.